### PR TITLE
test: SshnpdChannel

### PR DESCRIPTION
**Pull request by xavierchanth**
**- What I did**

- Tested SshnpdChannel, skipped listDevices, will return to it later, since it isn't as urgent as the other tests that we still need

**- How I did it**

**- How to verify it**

**- Description for the changelog**
test: SshnpdChannel

### CODE REVIEW

File: packages/noports_core/lib/src/common/mixins/async_initialization.dart [hunk 1]
```diff
@@ -10,7 +10,7 @@ mixin class AsyncInitialization {
   // * Public members
 
   /// Used to check if initialization has started
-  bool get initalizeStarted => _initializeStarted;
+  bool get initializeStarted => _initializeStarted;
```
Comment (xavierchanth) [inline packages/noports_core/lib/src/common/mixins/async_initialization.dart:14]: fixed a typo

File: packages/noports_core/lib/src/sshnp/util/sshnpd_channel/sshnpd_channel.dart [hunk 2]
```diff
@@ -57,11 +57,12 @@ abstract class SshnpdChannel with AsyncInitialization, AtClientBindings {
...
       regex: regex,
       shouldDecrypt: true,
-    ).listen(_handleSshnpdResponses);
+    ).listen(handleSshnpdResponses);
   }
 
   /// Main reponse handler for the daemon's notifications.
-  Future<void> _handleSshnpdResponses(AtNotification notification) async {
+  @visibleForTesting
+  Future<void> handleSshnpdResponses(AtNotification notification) async {
```
Comment (xavierchanth) [inline packages/noports_core/lib/src/sshnp/util/sshnpd_channel/sshnpd_channel.dart:65]: Scope: Private -> Visible for Testing

File: packages/noports_core/lib/src/sshnp/util/sshnpd_channel/sshnpd_channel.dart [hunk 3]
```diff
@@ -234,7 +235,8 @@ abstract class SshnpdChannel with AsyncInitialization, AtClientBindings {
   }
 
   /// A custom implementation of AtClient.getAtKeys which bypasses the cache
-  Future<List<AtKey>> _getAtKeysRemote(
+  @visibleForTesting
+  Future<List<AtKey>> getAtKeysRemote(
```
Comment (xavierchanth) [inline packages/noports_core/lib/src/sshnp/util/sshnpd_channel/sshnpd_channel.dart:239]: Scope: Private -> Visible for Testing

File: packages/noports_core/test/sshnp/util/sshrvd_channel/sshrvd_channel_test.dart [hunk 4]
```diff
@@ -17,7 +17,7 @@ void main() {
     late SshrvGeneratorStub<String> sshrvGeneratorStub;
     late MockAtClient mockAtClient;
     late StreamController<AtNotification> notificationStreamController;
-    late FunctionStub notifyStub;
+    late NotifyStub notifyStub;
     late SubscribeStub subscribeStub;
```
Comment (xavierchanth) [inline packages/noports_core/test/sshnp/util/sshrvd_channel/sshrvd_channel_test.dart:21]: Wasn't verifying the arguments passed into notify, so I created a NotifyStub class which allows me to validate those arguments

File: packages/noports_core/test/sshnp/util/sshrvd_channel/sshrvd_channel_test.dart [hunk 5]
```diff
@@ -50,25 +53,8 @@ void main() {
...
-              ..from = '@sshrvd'
-              ..to = '@client'
-              ..epochMillis = DateTime.now().millisecondsSinceEpoch
-              ..value = '$testIp,$portA,$portB',
-          );
-          notifyStub();
-        },
-        subscribe: ({regex, shouldDecrypt = false}) {
-          return subscribeStub();
-        },
```
Comment (xavierchanth) [inline packages/noports_core/test/sshnp/util/sshrvd_channel/sshrvd_channel_test.dart:71]: This removed function is now in a `when(notificationInvocation).thenAnswer(...)` block later on

File: packages/noports_core/test/sshnp/util/sshrvd_channel/sshrvd_channel_test.dart [hunk 6]
```diff
@@ -102,23 +88,55 @@ void main() {
...
+                      key.key == 'mydevice.${Sshrvd.namespace}' &&
+                      key.sharedBy == '@client' &&
+                      key.sharedWith == '@sshrvd' &&
+                      key.metadata != null &&
+                      key.metadata!.namespaceAware == false &&
+                      key.metadata!.ttl == 10000,
+                ),
+              ),
+              any(),
+            ),
```
Comment (xavierchanth) [inline packages/noports_core/test/sshnp/util/sshrvd_channel/sshrvd_channel_test.dart:139]: Here's the better argument matching as mentioned, checking that the values contained in the key passed in are as expected

File: packages/noports_core/test/sshnp/util/sshrvd_channel/sshrvd_channel_test.dart [hunk 7]
```diff
@@ -134,12 +152,27 @@ void main() {
...
-      await expectLater(stubbedSshrvdChannel.callInitialization(), completes);
+      await expectLater(stubbedSshrvdChannel.initialize(), completes);
 
       expect(stubbedSshrvdChannel.host, '234.234.234.234');
       expect(stubbedSshrvdChannel.port, 135);
     }); // test Initialization - non-sshrvd host
 
+    test('Initialization completes - sshrvd host', () async {
+      /// Set the required parameters
+      whenInitializationWithSshrvdHost();
```
Comment (xavierchanth) [inline packages/noports_core/test/sshnp/util/sshrvd_channel/sshrvd_channel_test.dart:163]: Added an init completion test, which checks if `completeInitialization()` is called when `callInitialization()` is called ... this only needs to be tested in certain places, usually impl classes, but initialization only occurs in the base class for sshrvd channel, so we can call and test for it in the base class.

**Review by gkc — APPROVED** [review]

**Pull request merged**